x-pack/legacy/plugins/maps/public/shared/layers/sources/ems_file_source/update_source_editor.js

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { TooltipSelector } from '../../../components/tooltip_selector';
+import { getEmsVectorFilesMeta } from '../../../../meta';
+
+export class UpdateSourceEditor extends Component {
+
+  static propTypes = {
+    onChange: PropTypes.func.isRequired,
+    tooltipProperties: PropTypes.arrayOf(PropTypes.string).isRequired
+  };
+
+  state = {
+    fields: null,
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this.loadFields();
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async loadFields() {
+    const emsFiles = await getEmsVectorFilesMeta();
+    const meta = emsFiles.find((source => source.id === this.props.layerId));
```

Comment:
Need to handle case where EMS layer id is not found. The source will display an error at the top of the editor so this function can just return early in that case